### PR TITLE
testing the UI outputs

### PR DESCRIPTION
**Pull request by Seif-Mamdouh**
Just testing out the expected outputs. 

### Issues with testcases
One of the issues I encountered was this testcase:

"A 10/21/2023 afternoon aa103 ee ee@rutgers.edu 30"
the expected output is "invalid location" 

our system prints out "invalid command"

Our system works fine if the user does insert a invalid Location. It just returns the wrong output, which is "invalid command". She expects "invalid location". Am I making sense?

the logic for the isValidLocation is inside location.java. That's where I need to need help. 

There's another testcase whether the department inserted is valid or not. I'm hope it has the same logic as location.

### Some fixes I have done
couple of boolean function if the user inserted the valid email, check if the event duration is between 30 to 120 min, etc.

added a testbed main for the date class and a pdf is attached to this project for the QA of the Dateclass

### CODE REVIEW

File: src/scheduler/Date.java [hunk 1]
```diff
@@ -102,17 +102,19 @@ public static Date parseDate(String dateStr) {
     private static final int[] dayInMonthConstant =
             {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
 
+
+    private static int min_month = 1;
```
Comment (mmm606) [inline src/scheduler/Date.java:105]: capitalize constants

File: src/scheduler/Date.java [hunk 2]
```diff
@@ -181,15 +183,10 @@ private long getMonthDifference(Calendar givenDate, Calendar endDate) {
         int endYear = endDate.get(Calendar.YEAR);
         int endMonth = endDate.get(Calendar.MONTH);
 
-        return (endYear - startYear) * MONTHS_IN_YEAR + (endMonth - startMonth);
+        return (long) (endYear - startYear) * MONTHS_IN_YEAR + (endMonth - startMonth);
```
Comment (mmm606) [inline src/scheduler/Date.java]: this needs to be a long?

File: src/scheduler/Date.java [hunk 3]
```diff
@@ -224,22 +221,40 @@ public String toString() {
...
+        Date date3 = new Date(2024, 4, 29);
+        Date date4 = new Date(2024, 2, 29);
+        Date date5 = new Date(2023, 13, 12);
+        Date date6 = new Date(2023, 0, 12);
+        Date date7 = new Date(2023, 2, 29);
+        Date date8 = new Date(-2023, 12, 2);
+
+        // Test the isValid method
+        System.out.println("date1 is : " + date1.isValid());
+        System.out.println("date2 is : " + date2.isValid());
```
Comment (mmm606) [inline src/scheduler/Date.java]: no assertions?

File: src/scheduler/EventOrganizer.java [hunk 4]
```diff
@@ -65,43 +66,62 @@ public void addEvent(String commandAdd) {
...
                 return;
             }
 
             Timeslot timeSlot =
                     Timeslot.valueOf(tokens[TIMESLOT_TOKEN_INDEX].toUpperCase());
-            Location location =
-                    Location.valueOf(tokens[LOCATION_TOKEN_INDEX].toUpperCase());
+            String locationToken = tokens[LOCATION_TOKEN_INDEX].toUpperCase();
+
+            if (Location.isValidLocation(locationToken)) {
```
Comment (mmm606) [inline src/scheduler/EventOrganizer.java]: isValidLocation returns true if not valid?

**Review by mmm606 — COMMENTED** [review]
Couple of comments let me take a look at the logic errors

File: src/scheduler/EventOrganizer.java (hunk 4 above)
Comment (Seif-Mamdouh) [reply src/scheduler/EventOrganizer.java]: Correct

Comment (mmm606) [reply src/scheduler/EventOrganizer.java]: this is the source of your location bug. Also get rid of the semi colon after the if statement's closing bracket

File: src/scheduler/Location.java [hunk 5]
```diff
@@ -26,6 +26,22 @@ public enum Location {
...
+    /**
+     * Checks if a given location string is valid by comparing it to the known building names
+     * in the Location enum.
+     *
+     * @param givenUserLocation The location string to validate.
+     * @return true if the location string is a valid building name, false otherwise.
+     */
+    public static boolean isValidLocation(String givenUserLocation) {
+        for (Location location : values()) {
+            if (location.buildingName.equalsIgnoreCase(givenUserLocation)) {
```
Comment (mmm606) [inline src/scheduler/Location.java]: You are comparing it to the full building name? That is the other source of your location bug.

**Pull request merged**